TYPO3.Neos/Documentation/HowTos/SelectingPageTemplate.rst

```diff
@@ -89,13 +89,85 @@ the `page` path by default. But you can add your own conditions to render a diff
+    employeePage = VendorName.VendorSite:EmployeePage
 
 This will now render the `employeePage` TypoScript path if a page of type `VendorName.VendorSite:Employee`
-is rendered on your website.
\ No newline at end of file
+is rendered on your website.
+
+Using a `DefaultPage` Prototype
+===============================
+
+This is an alternative and more flexible approach to the `Select Template based on NodeType` method descriped above.
+First we adjust the `default` `root` matcher not to render the page path, but a prototype derived from the current document node type name instead::
+
+    root {
+        default {
+            type = ${q(node).property('_nodeType') + '.Document'}
+            renderPath >
+        }
+    }
+
+Instead of simply defining our `page` object inside `root.ts2`, we create a new prototype based on a `page` prototype.
+The content will basically remain the same, make sure only to define bare essentials that all your future custom page types can profit from.
+
+Your basic `DefaultPage` prototype could look something like this::
+
+    prototype(VendorName:DefaultPage) < prototype(Page) {
+        head {
+            stylesheets {
+                site = TYPO3.TypoScript:Template {
+                    templatePath = 'resource//VendorName.VendorSite/Private/Templates/Includes/InlineStyles.html'
+                    sectionName = 'stylesheets'
+                }
+
+                mainStyle  = TYPO3.TypoScript:Tag {
+                    tagName = 'link'
+                    attributes {
+                        rel = 'stylesheet'
+                        href = TYPO3.TypoScript:ResourceUri {
+                            path = 'resource//VendorName.VendorSite/Public/Styles/Styles.css'
+                        }
+                    }
+                }
+            }
+        }
+        body {
+            templatePath = 'resource//VendorName.VendorSite/Private/Templates/Page/Default.html'
+            sectionName = 'body'
+        }
+    }
+
+Now we define our basic prototype for all `TYPO3.Neos.NodeTypes:Page` nodes.
+Since we extend `VendorName:DefaultPage` here, we can only define custom needs for `TYPO3.Neos.NodeTypes:Page` node types.
+
+For example::
+
+    prototype(TYPO3.Neos.NodeTypes:Page.Document) < prototype(VendorName:DefaultPage) {
+        body {
+            content {
+                main = PrimaryContent {
+                nodePath = 'main'
+                }
+            }
+        }
+    }
+
+All our custom document node types will be defined like this::
+
+    prototype(VendorName:Product.Document) < prototype(VendorName:DefaultPage) {
+        # custom properties for your node type
+    }
+
+In case we have a `layout` property within our node type configuration, we can define a prototype for this case too::
+
+    customLayout = TYPO3.Neos.NodeTypes:Page.Document {
+       # custom properties for your node type
+    }
+
+The credit for this approach goes to Christopher Hlubek.
```

Comment:
I don't know if we should add credits to docs, looks a bit weird for official docs, wdyt?
Maybe just mentioning @hlubek in a commit message would be enough?